numpy/lib/function_base.py

```diff
@@ -3792,16 +4027,135 @@ def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
                 else:
                     r = a.dtype.type(np.nan)
             else:
-                r[..., n.squeeze(0)] = a.dtype.type(np.nan)
+                r[n.squeeze(axis)] = a.dtype.type(np.nan)
         else:
             if r.ndim == 1:
                 r[:] = a.dtype.type(np.nan)
             else:
-                r[..., n.repeat(q.size, 0)] = a.dtype.type(np.nan)
+                r[..., n.squeeze(axis)] = a.dtype.type(np.nan)
 
     return r
 
 
+def _indices_for_weighted_quantile(ap, q, axis, weights):
```

Comment:
I have no idea what this function does. Can you add more to the docstring to explain the format of `ap`, and perhaps an example

Reply:
Will do.